### PR TITLE
Sample app to test library

### PR DESCRIPTION
**Pull request by mciechanowicz**

### CODE REVIEW

File: example/App.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+      const connectedDeviceIndex = devices.findIndex(device => device.id === connectedDevice.id)
+      const devicesArr = devices
+      devicesArr[connectedDeviceIndex] = { ...devicesArr[connectedDeviceIndex], isConnected: true }
+      setDevices(devicesArr)
+
+      const characteristics = await connectedDevice.discoverAllServicesAndCharacteristics()
+      console.log('Device services and characteristics: ', characteristics) 
+    } catch (error) {
+      console.log('Error! Connecting to device: ', error)
```
Comment (srgtuszy) [inline example/App.js:72]: We should communicate the errors to the user somehow, not just through the console

File: example/App.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  const handleConnectToDevice = async (deviceId) => {
+    handleStartLoading()
+    try {
+      const connectedDevice = await BLEmanager.connectToDevice(deviceId)
+      console.log('Connected to device: ', connectedDevice)
+
+      const connectedDeviceIndex = devices.findIndex(device => device.id === connectedDevice.id)
+      const devicesArr = devices
+      devicesArr[connectedDeviceIndex] = { ...devicesArr[connectedDeviceIndex], isConnected: true }
+      setDevices(devicesArr)
```
Comment (srgtuszy) [inline example/App.js:67]: Do we need an extra variable here?

File: example/App.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    } finally {
+      handleStopLoading()
+    }
+  }
+
+  const handleCancelConnection = async (deviceId) => {
+    handleStartLoading()
+    try {
+      const response = await BLEmanager.cancelDeviceConnection(deviceId)
+      console.log('Connection canceled succesfully: ', response)
```
Comment (srgtuszy) [inline example/App.js:82]: typo: `cancelled`

File: example/App.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+      const connectedDeviceIndex = devices.findIndex(device => device.id === response.id)
+      const devicesArr = devices
+      devicesArr[connectedDeviceIndex] = { ...devicesArr[connectedDeviceIndex], isConnected: false }
+      setDevices(devicesArr)
+    } catch (error) {
+      console.log('Error! Connection cancellation: ', error)
```
Comment (srgtuszy) [inline example/App.js:89]: We should have some kind of error handling mechanism

File: example/App.js [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+  const handleCancelConnection = async (deviceId) => {
+    handleStartLoading()
+    try {
+      const response = await BLEmanager.cancelDeviceConnection(deviceId)
+      console.log('Connection canceled succesfully: ', response)
+
+      const connectedDeviceIndex = devices.findIndex(device => device.id === response.id)
+      const devicesArr = devices
+      devicesArr[connectedDeviceIndex] = { ...devicesArr[connectedDeviceIndex], isConnected: false }
+      setDevices(devicesArr)
```
Comment (srgtuszy) [inline example/App.js:87]: This looks like a duplicated piece of code

File: example/app.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
+  "name": "reactNativeBLESampleApp",
```
Comment (srgtuszy) [inline example/app.json:2]: Maybe we can change the name to reflect the library name and our company?

**Review by srgtuszy — CHANGES_REQUESTED** [review]

File: example/App.js (hunk 2 above)
Comment (mciechanowicz) [reply example/App.js]: I changed it to make deep copy of state, it is needed because we don't want to mutate the state directly

**Pull request closed without merging**